### PR TITLE
Changes the `processModel` to generate `IList<Ix>` instead of `IList<x>`,

### PR DESCRIPTION
**Pull request by laura-rodriguez**

### CODE REVIEW

File: src/Okta.Sdk/Internal/CastingListAdapter{T}.cs [hunk 1]
```diff
@@ -54,6 +54,8 @@ private T Cast(object item)
             // Fall back to primitive conversion
             try
             {
+                if (!(item is IConvertible)) { return (T)item; }
```
Comment (laura-rodriguez) [inline src/Okta.Sdk/Internal/CastingListAdapter{T}.cs]: Disregard this line.

Comment (nbarbettini) [reply src/Okta.Sdk/Internal/CastingListAdapter{T}.cs]: Is this part of another PR?

Comment (laura-rodriguez) [reply src/Okta.Sdk/Internal/CastingListAdapter{T}.cs]: No, this was an attempt to handle the exception `the object doesn't implement the `IConvertible` interface`

**Comment by nbarbettini** [comment]
The problem was actually here: https://github.com/okta/okta-sdk-dotnet/pull/223/files#diff-8b7bc85bdb7632dc527405e6efdf2303L84

The factory didn't know how to "create" an interface. I added a lookup that will find the implementation `SamlAttributeStatement` given the interface `ISamlAttributeAttributeStatement`.

**Comment by laura-rodriguez** [comment]
Merged in #219

**Pull request closed without merging**